packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,152 @@
+import { Injectable, Scope } from 'graphql-modules';
+import { z } from 'zod';
+import * as Sentry from '@sentry/node';
+import { QueryAuditLogsArgs } from '../../../__generated__/types.next';
+import { User } from '../../../shared/entities';
+import { ClickHouse, sql } from '../../operations/providers/clickhouse-client';
+import { SqlValue } from '../../operations/providers/sql';
+import { Logger } from '../../shared/providers/logger';
+import { formatToClickhouseDateTime } from '../helpers';
+import { AuditLogEvent, auditLogSchema } from './audit-logs-types';
+
+export const AUDIT_LOG_CLICKHOUSE_OBJECT = z.object({
+  id: z.string(),
+  event_time: z.string(),
+  user_id: z.string(),
+  user_email: z.string(),
+  organization_id: z.string(),
+  event_action: z.string(),
+  metadata: z.string().transform(x => JSON.parse(x)),
```

Comment:
Is there no model for the metadata?

Reply:
I think it's just the transform, and then union discrimination validates it? 
Is there another way to implement that with zod?

Reply:
This can be fully modeled with zod types, something like this?


```suggestion
  metadata: z.string().transform(x => MyModel.parse(JSON.parse(x)),
```